.gitignore

```diff
@@ -1 +1,4 @@
 vendor
+composer.lock
```

Comment:
As this is a library i agree that there should be no composer.lock in the repository. For an application you always want a composer.lock in the repository which specifies your exact dependency's versions.